packages/@react-spectrum/listbox/docs/ListBox.mdx

```diff
@@ -28,35 +33,304 @@ category: Collections
+        selectionMode="multiple"
+        aria-label="Pick an animal"
+        items={options}
+        defaultSelectedKeys={['Bison', 'Koala']}
+        width="size-2400">
+        {item => <Item key={item.name}>{item.name}</Item>}
+      </ListBox>
+
+      <ListBox
+        selectionMode="multiple"
+        aria-label="Pick an animal"
+        items={options}
+        selectedKeys={animal}
+        onSelectionChange={selected => setAnimal(Array.from(selected))}
+        width="size-2400">
+        {item => <Item key={item.name}>{item.name}</Item>}
+      </ListBox>
+    </Flex>
+  );
+}
+```
+## Sections
+ListBox supports sections in order to group options. Sections can be used by wrapping groups of Items in a `Section` component. Each `Section` takes a `title` and `key` prop.
+
+### Static Items
+```tsx example
+<ListBox width="size-2400" aria-label="Pick your favorite" selectionMode="single">
+  <Section title="Animals">
+    <Item key="Aardvark">Aardvark</Item>
+    <Item key="Kangaroo">Kangaroo</Item>
+    <Item key="Snake">Snake</Item>
+  </Section>
+  <Section title="People">
+    <Item key="Danni">Danni</Item>
+    <Item key="Devon">Devon</Item>
+    <Item key="Ross">Ross</Item>
+  </Section>
+</ListBox>
+```
+### Dynamic Items
+Sections used with dynamic items are populated from a hierarchical data structure. Similarly to the props on `ListBox`, `Section` takes an array of data using the `items` prop.
+
+```tsx example
+function Example() {
+  let options = [
+    {name: 'Australian', children: [
+      {id: 2, name: 'Koala'},
+      {id: 3, name: 'Kangaroo'},
+      {id: 4, name: 'Platypus'}
+    ]},
+    {name: 'American', children: [
+      {id: 6, name: 'Bald Eagle'},
+      {id: 7, name: 'Bison'},
+      {id: 8, name: 'Skunk'}
+    ]}
+  ];
+
+  return (
+    <ListBox
+      aria-label="Pick an animal"
+      items={options}
+      selectionMode="single"
+      onSelectionChange={selected => console.log(selected)}
+      width="size-2400">
+      {item => (
+        <Section key={item.name} items={item.children} title={item.name}>
+          {item => <Item>{item.name}</Item>}
+        </Section>
+      )}
+    </ListBox>
+  );
+}
+```
 
 ## Events
+ListBox supports selection via mouse, keyboard, and touch. You can handle all of these via the `onSelectionChange`
+prop. ListBox will pass the selected `key` to the `onSelectionChange` handler.
+
+The following example uses an `onSelectionChange` handler to update the selection stored in React state.
+
+```tsx example
+function StaticExample() {
+  let [frequency, setFrequency] = React.useState();
+
+  return (
+    <>
+      <ListBox
+        aria-label="Choose frequency"
+        selectionMode="single"
+        onSelectionChange={selected => setFrequency(selected)}
+        width="size-2400">
+        <Item key="Rarely">Rarely</Item>
+        <Item key="Sometimes">Sometimes</Item>
+        <Item key="Always">Always</Item>
+      </ListBox>
+      <p>You selected: {frequency}</p>
```

Comment:
This is possibly a bit misleading. I think `frequency` here will actually be a Set of keys, not a single key. React seems to somehow convert it to a string, but we may want to be more explicit.